src/util/resourceValidationUtils.js

```diff
@@ -0,0 +1,143 @@
+const { resolveSchema } = require('@projecttacoma/node-fhir-server-core');
+const axios = require('axios');
+const { v4: uuidv4 } = require('uuid');
+const logger = require('../server/logger');
+
+/**
+ * Checks body of request by querying a FHIR validation server. Updates the submitted resource to
+ * include validated profiles in the meta.profile field. Responds with 400 if body is not valid FHIR
+ * 500 if validation server error
+ * @param {Object} req request object of an express request
+ * @param {Object} res response object of an express request
+ * @param {Function} next express function which passes info along after clearing the middleware
+ */
+async function validateFhir(req, res, next) {
```

Comment:
Nitpick, but if request type is not `PUT` or `POST`, we should be able to return `next()` early. Not crucial, but saves us a tiny morsel of computation

Reply:
actually, this comment might be irrelevant since this call should never happen if the request type is anything except PUT or POST, right? Since that's how the middleware is configured. Yeah, nvm I think you can ignore this!